sstables/sstables.cc

```diff
@@ -824,15 +826,15 @@ future<> sstable::read_toc() noexcept {
                 }
             }
             if (!_recognized_components.size()) {
-                throw malformed_sstable_exception("Empty TOC", filename(component_type::TOC));
+                throw_malformed_sstable_exception(sstlog, *this, "Empty TOC", filename(component_type::TOC));
             }
         });
     }).then_wrapped([this] (future<> f) {
         try {
             f.get();
         } catch (std::system_error& e) {
             if (e.code() == std::error_code(ENOENT, std::system_category())) {
-                throw malformed_sstable_exception(filename(component_type::TOC) + ": file not found");
+                throw_malformed_sstable_exception(sstlog, *this, format("{}", std::current_exception()), filename(component_type::TOC));
```

Comment:
Changing the log message should probably be done in its own commit.

Reply:
done in v2